build-aux/snap/snapcraft.yaml

```diff
@@ -28,172 +29,305 @@ license: GPL-3.0
       # the current build environment does not support them
       # For some reason, some snapcraft version remove the "build-aux" folder
       # and move the contents up when the data is uploaded; this conditional
       # manages it.
-      if [ -d "$SNAPCRAFT_PROJECT_DIR/build-aux" ]; then
-        cp $SNAPCRAFT_PROJECT_DIR/build-aux/snap/local/apparmor/af_names.h .
+      if [ -d "${CRAFT_PROJECT_DIR}/build-aux" ]; then
+        cp "${CRAFT_PROJECT_DIR}"/build-aux/snap/local/apparmor/af_names.h .
       else
-        cp $SNAPCRAFT_PROJECT_DIR/snap/local/apparmor/af_names.h .
+        cp "${CRAFT_PROJECT_DIR}"/snap/local/apparmor/af_names.h .
       fi
-      make -j$(nproc)
-      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd
-      cp -a apparmor_parser $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/
-      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor
-      cp -a parser.conf $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor/
-      cd $SNAPCRAFT_PART_BUILD/profiles
-      make -j$(nproc)
-      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d
-      cp -a apparmor.d/abi $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
-      cp -a apparmor.d/abstractions $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
-      cp -a apparmor.d/tunables $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+      install -Dm755 -t "${CRAFT_PART_INSTALL}/usr/lib/snapd" apparmor_parser
+      install -Dm644 -t "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor" parser.conf
+      cd "${CRAFT_PART_BUILD}/profiles"
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+      mkdir -p "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+      cp -a apparmor.d/abi "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+      cp -a apparmor.d/abstractions "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+      cp -a apparmor.d/tunables "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+    prime:
+      # We are statically linked
+      - -usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/**
+      - -usr/include
+    <<:
+      - *dynamic-linker
+    override-prime: |
+      craftctl default
+      python3 "${CRAFT_PROJECT_DIR}/build-aux/snap/local/patch-dl.py" "/snap/snapd/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/${DYNAMIC_LINKER}"
+
+  snapd:
+    plugin: nil
+    source: .
+    build-snaps:
+      - go/1.18/stable
+    after:
+      - apparmor
+    # these packages are needed to call mkversion.sh in override-pull, all other
+    # dependencies are installed using apt-get build-dep
+    build-packages:
+      - git
+      - dpkg-dev
+      - autoconf
+      - autoconf-archive
+      - automake
+      - xfslibs-dev
+      - libudev-dev
+      - libcap-dev
+      - libseccomp-dev
+      - pkg-config
+    override-pull: |
+      craftctl default
+      # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
+      craftctl set version="$(./mkversion.sh --output-only)"
+      ./get-deps.sh --skip-unused-check
```

Comment:
google:ubuntu-22.04-64:tests/main/snapd-snap:destructive failed, likely caused by new git checks directory ownership:

```
 :: Obtaining c-dependencies
:: fatal: detected dubious ownership in repository at '/home/gopath/src/github.com/snapcore/snapd/parts/snapd/src/c-vendor/squashfuse'
:: To add an exception for this directory, call:
::
::   git config --global --add safe.directory /home/gopath/src/github.com/snapcore/snapd/parts/snapd/src/c-vendor/squashfuse
:: fatal: detected dubious ownership in repository at '/home/gopath/src/github.com/snapcore/snapd/parts/snapd/src/c-vendor/squashfuse'
```

Reply:
I suppose we already called `get-deps.sh` before calling snapcraft.

Reply:
I have fixed the test. It now removes `c-vendor/squashfuse` before calling snapcraft in destructive mode.